odoo/fields.py

```diff
@@ -2502,23 +2502,47 @@ def write(self, records, value):
         dirty = self.column_type and self.store and any(records._ids)
         records.env.cache.update(records, self, itertools.repeat(cache_value), dirty=dirty)
 
-    def _image_process(self, value):
+    def _image_process(self, value, records=None):
         if self.readonly and not self.max_width and not self.max_height:
             # no need to process images for computed fields, or related fields
             return value
         try:
             img = base64.b64decode(value or '') or False
         except:
             raise UserError(_("Image is not encoded in base64."))
+
+        if img and records and guess_mimetype(img, '') == 'image/webp':
+            if not self.max_width and not self.max_height:
+                return value
+            # Fetch resized version.
+            Attachment = records.env['ir.attachment']
+            checksum = Attachment._compute_checksum(img)
+            origin = Attachment.search([
+                ['id', '!=', False],  # No implicit condition on res_field.
+                ['checksum', '=', checksum],
+            ])
+            if origin:
+                origin_ids = [attachment.id for attachment in origin]
+                resized_domain = [
+                    ['id', '!=', False],  # No implicit condition on res_field.
+                    ['res_model', '=', 'ir.attachment'],
+                    ['res_id', 'in', origin_ids],
+                    ['description', '=', 'resize: %s' % max(self.max_width, self.max_height)],
+                ]
+                resized = Attachment.sudo().search(resized_domain, limit=1)
+                if resized:
+                    return resized.datas or value
+                    # Fallback on non-resized image.
+
         return base64.b64encode(image_process(img,
             size=(self.max_width, self.max_height),
             verify_resolution=self.verify_resolution,
         ) or b'') or False
 
-    def _process_related(self, value):
+    def _process_related(self, value, record):
```

Comment:
This new param in `_process_related` and `_image_process` is just to be used through this line:
```py
Attachment = records.env['ir.attachment']
```
Why do you need to find the `env` that way? Or is it just to be used as a boolean in the line `if img and records and guess_mimetype(img, '') == 'image/webp':`? In this case, it should just be a Boolean.
And maybe the `and records` part of the condition which this PR adds should be commented to explain why we only do that if we have records

Reply:
>Why do you need to find the env that way?

It is because I am within a `Field` of `fields.py`, `self` is not a model: I do not know how I could get access to an `env` from there. Other methods in there rely on a model or records parameter to access one.
Now sending `records.env` from all calls to make it more clear.